### PR TITLE
incrementally autoupdate tldr pages

### PR DESCRIPTION
**Pull request by invakid404**
Hi there 👋🏻,

First things first, sorry for the massive diff -- had to bring this repo in sync with latest tldr upstream for the stuff in this PR to work :D

As a person who uses tldr pages on a daily basis and as a person who's trying to introduce Dendron in their workflow, I immediately liked the idea of this repo. I saw that it's not really up to speed with the latest tldr-pages commits though, so I thought I might as well automate it :^)

This PR enhances the existing publish workflow to automatically determine whether any pages have updated upstream and regenerates and commits them. I've achieved this by making tldr-pages a submodule of this repo, because that way it remembers its latest commit, which I use as an indicator of the diff. The workflow retrieves the cached tldr-pages commit, tries to update it, looks at the new commit, compares the two and comes up with a list of markdown files which have changed. If no files have changed between the two commits, nothing will change. It is currently set up to run every hour.

I've tried my best to optimize it in terms of imported changes in each run of the workflow. Since Dendron imports all of the pages at once and inherently changes their date created and date modified, I've made sure to update stuff in an incremental manner. Only the pages that have changed between the old and new commits will be automatically committed and pushed to the main branch.

Currently, the GitHub Pages website is redeployed every run of the workflow and isn't updated incrementally, but I think it's reasonable since the more critical part of it is to avoid forcing users of this vault to pull in garbage changes all the time, redeploying GH Pages isn't a massive deal and was the behavior before this change anyway.

Hope you find these changes useful :)

### CODE REVIEW

**Comment by invakid404** [comment]
BTW, the massive commit you probably want to avoid when looking at this is deadcd5051e5d6492e8047e7e1798611e4ea2b67 :D

**Comment by invakid404** [comment]
After fixing the regex, it appears to be working perfectly. Commit 1319ed2666da4685d6baeb5c46c0d935efed91e4 updates the imported Dendron notes for [these tldr commits](https://github.com/tldr-pages/tldr/compare/0e9bf6b396502db16df0241698409022f67cfbf5...3762d44bb2e260835837438577b8ec0cf6924063). As you can see, it correctly updated all the pages in English and ignored all the translation changes.

File: .github/workflows/dendron-action.yml [hunk 1]
```diff
@@ -12,17 +15,79 @@ jobs:
...
+    - name: Retrieve latest commit
+      id: commit
+      if: steps.updated-pages.outputs.FILES != ''
+      run: |
+        echo "::set-output name=SHA::$(git rev-parse HEAD)"
 
     - name: Build pod
-      run: npm run dendron-cli -- buildSiteV2 --wsRoot .  --stage prod
+      if: steps.updated-pages.outputs.FILES != ''
+      run: yarn dendron-cli buildSiteV2 --wsRoot .  --stage prod
```
Comment (kevinslin) [inline .github/workflows/dendron-action.yml:80]: heads up that this is deprecated and will be removed in future versions. will be updating this when we remove it so no action needed on your part but just FYI

**Review by kevinslin — APPROVED** [review]

**Comment by kevinslin** [comment]
> Hi there 👋🏻,
> 
> First things first, sorry for the massive diff -- had to bring this repo in sync with latest tldr upstream for the stuff in this PR to work :D
> 
> As a person who uses tldr pages on a daily basis and as a person who's trying to introduce Dendron in their workflow, I immediately liked the idea of this repo. I saw that it's not really up to speed with the latest tldr-pages commits though, so I thought I might as well automate it :^)
> 
> This PR enhances the existing publish workflow to automatically determine whether any pages have updated upstream and regenerates and commits them. I've achieved this by making tldr-pages a submodule of this repo, because that way it remembers its latest commit, which I use as an indicator of the diff. The workflow retrieves the cached tldr-pages commit, tries to update it, looks at the new commit, compares the two and comes up with a list of markdown files which have changed. If no files have changed between the two commits, nothing will change. It is currently set up to run every hour.
> 
> I've tried my best to optimize it in terms of imported changes in each run of the workflow. Since Dendron imports all of the pages at once and inherently changes their date created and date modified, I've made sure to update stuff in an incremental manner. Only the pages that have changed between the old and new commits will be automatically committed and pushed to the main branch.
> 
> Currently, the GitHub Pages website is redeployed every run of the workflow and isn't updated incrementally, but I think it's reasonable since the more critical part of it is to avoid forcing users of this vault to pull in garbage changes all the time, redeploying GH Pages isn't a massive deal and was the behavior before this change anyway.
> 
> Hope you find these changes useful :)

This is awesome, first - thank you for the PR and apologies about the long latency in responding. 

> Since Dendron imports all of the pages at once and inherently changes their date created and date modified, I've made sure to update stuff in an incremental manner.

Ah, this can be updated. Feel free to send a feature request our way to preserve original timestamps

**Pull request merged**